coalib/testing/BaseTestHelper.py

```diff
@@ -0,0 +1,36 @@
+from coala_utils.Comparable import Comparable
+
+
+class BaseTestHelper:
+    """
+    This is a base class for all Bears' tests of coala's testing API.
+    """
+
+    def assertComparableObjectsEqual(self, observed_result, expected_result):
+        """
+        Compares individual fields of the Result object and yields messages in
+        case of an attribute mismatch.
+        """
+        if len(observed_result) == len(expected_result):
+            messages = ''
+            for observed, expected in zip(observed_result, expected_result):
+                if (isinstance(observed, Comparable)
+                        and isinstance(expected, Comparable)) \
+                        and (type(observed) is type(expected)):
+                    for attribute in type(observed).__compare_fields__:
+                        try:
+                            assert getattr(observed, attribute) == \
+                                getattr(expected, attribute), \
+                                '{} mismatch: {}, {} != {}, {}'.format(
+                                    attribute,
+                                    observed.origin, observed.message,
+                                    expected.origin, observed.message)
+                        except AssertionError as ex:
+                            messages += (str(ex) + '\n\n')
+                else:
+                    assert observed_result == expected_result, '%s != %s' % \
+                        (observed_result, expected_result)
+            if messages:
+                raise AssertionError(messages)
+        else:
```

Comment:
You can write maybe like 
`if not len(observed_result) == len(expected_result):` This would eliminate one level of indentation atleast. There are so many nested if and for loops.